### ISSUE TITLE
Add support for TPU devices

### ISSUE THREAD

**Issue by vfdev-5** [pytorch-ignite/code-generator#173]
### Clear and concise description of the problem

It would be good to provide an option to select accelerator as TPU instead of GPU
We can also auto-select TPU accelerator if open with Colab + add torch_xla installation steps.

**Comment by afzal442** [comment]
hi there, trying to refactor it with a little digging into this following this way https://colab.research.google.com/github/pytorch/xla/blob/master/contrib/colab/mnist-training.ipynb#scrollTo=0WzFpOKDmURO

We can allow it with this nature;
cc @vfdev-5

**Comment by afzal442** [comment]
something like this as follow;
![image](https://user-images.githubusercontent.com/11625672/135754581-5031301e-1fdf-4f91-8e63-bb9667aade30.png)

**Comment by afzal442** [comment]
// @ydcjeff

**Comment by vfdev-5** [comment]
@sayantan1410 I updated issue description adding few initial steps on how I would tackle this issue.

**Comment by sayantan1410** [comment]
@vfdev-5 I will start to work as per the description, and will let you know if I face some problem.

**Comment by sayantan1410** [comment]
> 0) Try a template with TPUs. Choose distributed training option with 8 processes and spawning option. "Open in colab" one template, for example, vision classification template, install manually torch_xla (see https://colab.research.google.com/drive/1E9zJrptnLJ_PKhmaP5Vhb6DTVRvyrKHx) and run the code with backend `xla-tpu`: `python main.py --nproc_per_node 8 --backend nccl`. If everything is correctly done, training should probably run

@vfdev-5 Hello, should i do this from the code-generator website or by running it locally or both works ?

**Comment by vfdev-5** [comment]
> > 0. Try a template with TPUs. Choose distributed training option with 8 processes and spawning option. "Open in colab" one template, for example, vision classification template, install manually torch_xla (see https://colab.research.google.com/drive/1E9zJrptnLJ_PKhmaP5Vhb6DTVRvyrKHx) and run the code with backend `xla-tpu`: `python main.py --nproc_per_node 8 --backend nccl`. If everything is correctly done, training should probably run
> 
> @vfdev-5 Hello, should i do this from the code-generator website or by running it locally or both works ?

From code-generator and exporting in colab as I explained. You can't test that locally as you should use TPUs.

**Comment by sayantan1410** [comment]
@vfdev-5 Got it, Thank you.

**Comment by sayantan1410** [comment]
@vfdev-5 Hello, I am facing a issue to run the colab notebook, I tried to install the torch_xla manually and then start the training, Here's the link to the notebook - https://colab.research.google.com/drive/15tlo1Js4vCXSDB5yqLQJ9byvwvtoEuEU?usp=sharing. The main problem that I am facing is that whenever I am running '!pip install -r requirements.txt' it is uninstalling the torch 1.9 version and reinstalling another version. But without installing the requirements.txt also it is not working. 

Can you please check it out and let me know, what I am missing.

**Comment by vfdev-5** [comment]
@sayantan1410 can you please update your colab and show where you call `!pip install -r requirements.txt` and the output it gives. By the way, I also forgot in the description to mention that we need to set accelerator as TPU.

**Comment by sayantan1410** [comment]
@afzal442 Okay I will try to remove the two from the requirements.txt and try. And setting the accelerator to TPU was written in the colab notebook that you linked in the description.

**Comment by sayantan1410** [comment]
@vfdev-5 Hey, I was trying to change the requirements.txt from here
![Screenshot (12)](https://user-images.githubusercontent.com/56483183/148329218-54b466cc-d09e-400e-ad19-8a3ca813a10b.png)
But I cannot edit this. So I tried to install the other libraries manually, but the same problem is persisting. The colab link is same as the previous. Let me know if there is some other way to change the requirements.txt .

**Comment by vfdev-5** [comment]
@sayantan1410 the issue in colab is not with the dependencies but with the way you start trainings. Please read ignite docs on `idist.Parallel` and also see the step 1 and 2 of this issue description.

**Comment by sayantan1410** [comment]
@vfdev-5 Hey I tried to run the code but getting a "Aborted: Session 96f4ae2c056673d1 is not found" error. Can I start working on the UI in the meantime while I am trying to solve the colab issue ?

**Comment by vfdev-5** [comment]
> Can I start working on the UI in the meantime while I am trying to solve the colab issue ?

yes, that's the final goal of the issue. The work with colab is a step 0 to check if things could work.

**Comment by sayantan1410** [comment]
@vfdev-5 > "Aborted: Session 96f4ae2c056673d1 is not found"

Can you please check once why is this coming ?

**Comment by vfdev-5** [comment]
Looks like an internal issue with TPUs on Colab, try to do "factory reset runtime" and see if the issue persists. If you have a Kaggle account, you can also check the same code on their TPUs.

**Comment by sayantan1410** [comment]
@vfdev-5 I tried doing "factory reset runtime", but that did not work. I will try running it on Kaggle notebooks.
Also for the UI update part, I have done something like this -
![Screenshot (14)](https://user-images.githubusercontent.com/56483183/148495655-cd066a8e-fb93-4b00-af59-612b8eab13f4.png)
Should I populate the dropdown by creating "backend.json" and then calling it in " TabTemplates.vue ", Or there is a better way ? 
Also what is the next step ?
